webservice/soap/include/inc.soap_functions.php

```diff
@@ -1063,6 +1067,37 @@ public static function processBackgroundTask($sid, $task_id) {
 		return $bg->processBackgroundTask($sid, $task_id);
 	}
 
-}
-
-?>
+	/**
+	 * @param string $name
+	 * @param array $arguments
+	 * @return mixed
+	 * @throws SoapFault
+	 */
+	public function __call($name, $arguments) {
+		// SoapHookPlugins need the client-ID submitted
+		if (!isset($_GET['client_id'])) {
+			throw new SoapFault('SOAP-ENV:Server', "Function '{$name}' does not exist");
+		}
+		// Note: We need to bootstrap ILIAS in order to get $ilPluginAdmin and load the soap plugins.
+		// We MUST use a context that does not handle authentication at this point (session is checked by SOAP).
+		// Solution: Use the REST context
+		ilContext::init(ilContext::CONTEXT_REST);
```

Comment:
Please can you create a new context for the first initialisation. Reusing a different context is to error prone.